include/class.attachment.php

```diff
@@ -107,16 +107,24 @@ function getType() { return $this->key['type']; }
      * additionally, add new files whose IDs are in the list provided.
      */
     function keepOnlyFileIds($ids, $inline=false, $lang=false) {
+        $attachments = array();
         if (!$ids) $ids = array();
-        $new = array_fill_keys($ids, 1);
+        $new = array_flip($ids);
         foreach ($this as $A) {
             if (!isset($new[$A->file_id]) && $A->lang == $lang && $A->inline == $inline)
                 // Not in the $ids list, delete
                 $this->remove($A);
             unset($new[$A->file_id]);
         }
+        // Format $new for upload() with new name
+        foreach ($new as $id=>$name) {
```

Comment:
Move `$attachments = array();` from line 110 down here... unit close to block of code that uses the variable to avoid overwrite errors as code get added in the future.